src/BadiNodaTimeTest/YearInfoTests.cs

```diff
@@ -103,5 +103,17 @@ public void HolyDays_Other()
       hd2.TimeCode.ShouldEqual(SpecialTimeCode.H01);
       hd2.DayCode.ShouldEqual(HolyDayCode.AscAbdul);
     }
+
+    [Fact]
+    public void GetSpecialDays_HolyDaysWanted()
+    {
+        var yearInfo = new BadiYearInfo(174);
+
+        var specialDaysList = yearInfo.GetSpecialDays(SpecialDayType.HolyDay_WorkSuspended, HolyDayCode.NawRuz);
+        specialDaysList.Count().ShouldEqual(1);
+
+        specialDaysList = yearInfo.GetSpecialDays(SpecialDayType.HolyDay_WorkSuspended, HolyDayCode.Ridvan1);
+        specialDaysList.Count().ShouldEqual(1);
+    }
```

Comment:
```suggestion
    [Fact]
    public void GetSpecialDays_HolyDaysWanted()
    {
        var yearInfo = new BadiYearInfo(174);

        var specialDaysList = yearInfo.GetSpecialDays(SpecialDayType.HolyDay_WorkSuspended, HolyDayCode.NawRuz);
        specialDaysList.Count().ShouldEqual(1);

        specialDaysList = yearInfo.GetSpecialDays(SpecialDayType.HolyDay_WorkSuspended, HolyDayCode.Ridvan1);
        specialDaysList.Count().ShouldEqual(1);
    }
```

Reply:
In just reading the code change proposed, I can't be confident that it is working fully. I'd need to run the code and do a number of other tests to be sure.

So it would be good if you could add more tests and made them more comprehensive. For instance, it would be good to also test calls that will give no result, like `.GetSpecialDays(SpecialDayType.HolyDay_Other, HolyDayCode.Ridvan1)`.